File: src/main/java/com/lsd/events/Newpage.java
package com.lsd.events;

import lombok.Value;

import static java.lang.System.lineSeparator;
import static org.apache.commons.lang3.StringUtils.isBlank;

/**
 * Used to indicate that a new page should be created in a sequence diagram.
 */
@Value(staticConstructor = "titled")
public class Newpage implements SequenceEvent {
    String keyword = "newpage";
    String title;

    public String toMarkup() {
        String titleValue = isBlank(title) ? "" : " '" + title + "'";
        return lineSeparator() + keyword + titleValue + lineSeparator();
    }
}

File: src/main/kotlin/com/lsd/events/Newpage.kt
package com.lsd.events

import java.lang.System.lineSeparator

/**
 * Used to indicate that a new page should be created in a sequence diagram.
 */
class Newpage(val pageTitle: PageTitle) : SequenceEvent {
    private val keyword = "newpage"

    override fun toMarkup(): String {
        val titleMarkup = pageTitle.title.let { if (it.isEmpty()) "" else " '$it'" }
        return "${lineSeparator()}$keyword$titleMarkup${lineSeparator()}"
    }

    companion object {
        @JvmStatic
        fun titled(title: String): Newpage = Newpage(PageTitle(title))
    }
}